include/mlir/Dialect/StandardOps/Ops.td

```diff
@@ -836,6 +836,35 @@ def ShlISOp : IntArithmeticOp<"shlis"> {
   let summary = "signed integer shift left";
 }
 
+def SplatOp : Std_Op<"splat", [NoSideEffect]> {
```

Comment:
This op should use the TCresVTEtIsSameAsOp<0, 0> trait

Reply:
The result and the operand doesn't have the same element type here. The operand type here is the element type of the result.

Reply:
TCresVTEtIsSameAsOp supports this exact use case: https://github.com/tensorflow/mlir/blob/c49b963db45a8425371083dc3ddef860bd28ad5c/include/mlir/IR/OpBase.td#L1517

Reply:
But the doc comment over there says it checks if the operand and result have the same element type, which isn't the case here. Did you mean:
 TCOpIsBroadcastableToRes ?

Reply:
Ah yes, I see I was thinking of just using TCresVTEtIsSameAsOpBase. 
https://github.com/tensorflow/mlir/blob/be6746fc3b9e6bd0527cc961256e362f44130cd4/include/mlir/Dialect/VectorOps/VectorOps.td#L53

TCOpIsBroadcastableToRes won't work because both need to be shaped.

Reply:
Sorry, I still don't understand your comment. Those traits appear to be checking whether either their self types or elemental types match, which isn't what we need for splat. As an aside, the doc comments for both (TCresVTEtI..., TCOpIsBroad...) of these are inconsistent with their behavior and their names.

Reply:
I'm trying to see if we can compose the traits together, i.e. selfOrElementType is too broad for what we really want. It allows shaped/non-shaped types on both sides, but that should be fine given that we constrain the operand and result types as well. That being said, we should likely add a trait that does exactly what we want given that we have the same verification for ExtractElementOp(Doesn't have to be this commit, but an issue/TODO would be good).

Reply:
Yes, it would be good to have a trait. But note that ExtractElementOp has the reverse requirement - the result there is an element of the operand, and the latter is the shaped type.

Reply:
Yes, thanks for the correction. I was thinking InsertElement in my head.

Reply:
Added a TODO at the verify method.